scalingo: init at 1.29.1

cimm opened the pull request:
###### Description of changes

Scalingo is a French PaaS, think Heroku or Fly.io. They have a CLI similar to the Heroku one to interact with their service: set environment variables, console access to databases, etc. It's written in Go and supports Linux, macOS, Windows, FreeBSD and OpenBSD. More information can be found in the [Scalingo CLI documentation](https://doc.scalingo.com/platform/cli/start).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cimm committed:
scalingo: init at 1.28.2

cimm committed:
Run tests when building

cimm committed:
Fix DISABLE_UPDATE_CHECKER environment variable

cimm commented:
Anything else I need to do/can do to get this pull request merged? Thank you!

josephpage commented on pkgs/development/tools/scalingo/default.nix:
```diff
@@ -0,0 +1,33 @@
+  meta = with lib; {
+    description = "Command Line client for Scalingo PaaS";
+    homepage = "https://doc.scalingo.com/platform/cli/start";
+    changelog = "https://github.com/Scalingo/cli/blob/master/CHANGELOG.md#1282";
```
Je pense que vous ne devriez pas faire pointer vers une ligne en particulier, vous devriez enlever la balise `#1282` à la fin de l'url.

cimm replied:
I removed the tag. Thanks for the feedback!

cimm committed:
Remove anchor from changelog URL

cimm committed:
scalingo: 1.28.2 -> 1.29.1

cimm committed:
Merge branch 'NixOS:master' into add-scalingo-package

cimm commented:
This pull request has been pending for a few months now. I assume the maintainers simply have too much work on their plate.

Is there anything I can do to get this merged?

nlewo reviewed (changes requested):
Could you remove the `/bin/dists` file (don't know what it is) in a `postInstall` phase?

Also, this package should follow the new packge location convention. Could you move this package to `pkgs/by-name/sc/scalingo/` as described in https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package

cimm committed:
Merge branch 'master' into add-scalingo-package

cimm committed:
Fix typos in package description

cimm committed:
Remove ldflags

cimm commented on pkgs/development/tools/scalingo/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  vendorHash = null;
+
+  ldflags = [
```
I thought they resulted in a smaller binary but will remove them if they are not needed.

cimm committed:
Update notation

cimm committed:
Use new packge location convention

cimm committed:
Remove unused dists binary post install

cimm commented:
@nlewo Thanks for the feedback! Anything else that needs to be done for this package?

- removed the unnecessary `dists` binary in a `postInstall` hook
- moved the package to the new package location
- removed the `ldflags`
- updated your naming suggestions and typos in the description

nlewo reviewed (approved):
@cimm Thx for your changes!
